### PR TITLE
HTS Client Card Export Report 2019

### PR DESCRIPTION
**Pull request by dbaluku**
https://app.asana.com/0/0/1133167201255018/f

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/ugandaemrreports/reports/SetUpHTSClientCardDataExportReport.java [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+		l.add(df.getStartDateParameter());
+		l.add(df.getEndDateParameter());
+		return l;
+	}
+	
+	@Override
+	public List<ReportDesign> constructReportDesigns(ReportDefinition reportDefinition) {
+		List<ReportDesign> l = new ArrayList<ReportDesign>();
+		l.add(buildReportDesign(reportDefinition));
+//		l.add(buildExcelReportDesign(reportDefinition));
```
Comment (ssmusoke) [inline api/src/main/java/org/openmrs/module/ugandaemrreports/reports/SetUpHTSClientCardDataExportReport.java]: @dbaluku delete any unused code

File: api/src/main/java/org/openmrs/module/ugandaemrreports/reports/SetUpHTSClientCardDataExportReport.java [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+	}
+	
+	@Override
+	public String getUuid() {
+		return "96e0926d-1606-4de6-943f-cb036bdc15ad";
+	}
+	
+	@Override
+	public String getVersion() {
+		return "0.1.6";
```
Comment (ssmusoke) [inline api/src/main/java/org/openmrs/module/ugandaemrreports/reports/SetUpHTSClientCardDataExportReport.java]: @dbaluku Start with 3.0.0 for all new HMIS reports

File: api/src/main/java/org/openmrs/module/ugandaemrreports/reports/SetUpHTSClientCardDataExportReport.java (hunk 1 above)
Comment (dbaluku) [reply api/src/main/java/org/openmrs/module/ugandaemrreports/reports/SetUpHTSClientCardDataExportReport.java]: Done

File: api/src/main/java/org/openmrs/module/ugandaemrreports/reports/SetUpHTSClientCardDataExportReport.java (hunk 2 above)
Comment (dbaluku) [reply api/src/main/java/org/openmrs/module/ugandaemrreports/reports/SetUpHTSClientCardDataExportReport.java]: Done

File: api/src/main/java/org/openmrs/module/ugandaemrreports/reports/SetUpHTSClientCardDataExportReport.java [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+import org.openmrs.module.ugandaemrreports.reporting.metadata.Dictionary;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Health Unit TB Register Report
```
Comment (ssmusoke) [inline api/src/main/java/org/openmrs/module/ugandaemrreports/reports/SetUpHTSClientCardDataExportReport.java]: @dbaluku Change the class description

File: api/src/main/java/org/openmrs/module/ugandaemrreports/reports/SetUpHTSClientCardDataExportReport.java [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+
+	@Autowired
+	SharedDataDefintion sdd;
+
+	@Autowired
+	private BuiltInPatientDataLibrary builtInPatientData;
+	
+	@Override
+	public String getDescription() {
+		return "HTS Client Card Data Export" ;
```
Comment (ssmusoke) [inline api/src/main/java/org/openmrs/module/ugandaemrreports/reports/SetUpHTSClientCardDataExportReport.java]: @dbaluku add "for 2019 version of HTS Client card"

File: api/src/main/java/org/openmrs/module/ugandaemrreports/reports/SetUpHTSClientCardDataExportReport.java [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+		dsd.addColumn("Pre-test Conselling Done",sdd.definition("pre-testConseling",  getConcept("193039f1-c378-4d81-bb72-653b66c69914")), "onOrAfter=${startDate},onOrBefore=${endDate}", new ObsDataConverter());
+		dsd.addColumn("Counseled As", sdd.definition("counseled as",  getConcept("b92b1777-4356-49b2-9c83-a799680dc7d4")), "onOrAfter=${startDate},onOrBefore=${endDate}", new ObsDataConverter());
+		dsd.addColumn("HIV Test Results", sdd.definition("HIVresults",  getConcept("3d292447-d7df-417f-8a71-e53e869ec89d")), "onOrAfter=${startDate},onOrBefore=${endDate}", new ObsDataConverter());
+		dsd.addColumn("Syphillis Duo Results",sdd.definition("syphillis duo results",  getConcept("16091701-69b8-4bc7-82b3-b1726cf5a5df")), "onOrAfter=${startDate},onOrBefore=${endDate}", new ObsDataConverter());
+		dsd.addColumn("Recency Results", sdd.definition("recency Results",  getConcept("141520BBBBBBBBBBBBBBBBBBBBBBBBBBBBBB")), "onOrAfter=${startDate},onOrBefore=${endDate}", new ObsDataConverter());
```
Comment (ssmusoke) [inline api/src/main/java/org/openmrs/module/ugandaemrreports/reports/SetUpHTSClientCardDataExportReport.java]: @dbaluku Field missing - Consented for Recency Testing

File: api/src/main/java/org/openmrs/module/ugandaemrreports/reports/SetUpHTSClientCardDataExportReport.java (hunk 4 above)
Comment (dbaluku) [reply api/src/main/java/org/openmrs/module/ugandaemrreports/reports/SetUpHTSClientCardDataExportReport.java]: done

File: api/src/main/java/org/openmrs/module/ugandaemrreports/reports/SetUpHTSClientCardDataExportReport.java (hunk 3 above)
Comment (dbaluku) [reply api/src/main/java/org/openmrs/module/ugandaemrreports/reports/SetUpHTSClientCardDataExportReport.java]: done

File: api/src/main/java/org/openmrs/module/ugandaemrreports/reports/SetUpHTSClientCardDataExportReport.java (hunk 5 above)
Comment (dbaluku) [reply api/src/main/java/org/openmrs/module/ugandaemrreports/reports/SetUpHTSClientCardDataExportReport.java]: done

**Pull request merged**